packages/iris-grid/src/IrisGridTheme.ts

```diff
@@ -61,36 +61,61 @@ export interface IrisGridDensity {
   filterBarHeight: number;
 }
 
-export const NORMAL_DENSITY_THEME = {
-  cellHorizontalPadding: 5,
-  headerHorizontalPadding: 12,
-  minColumnWidth: 55,
-  rowHeight: 19,
-  font: '12px Fira Sans, sans-serif',
-  headerFont: '600 12px Fira Sans, sans-serif',
-  iconSize: 16,
-  columnHeaderHeight: 30,
-  filterBarHeight: 30,
-} satisfies IrisGridDensity;
-
-export const COMPACT_DENSITY_THEME = {
-  cellHorizontalPadding: 2,
-  headerHorizontalPadding: 10,
-  minColumnWidth: 10,
-  rowHeight: 16,
-  font: '11px Fira Sans, sans-serif',
-  headerFont: '600 11px Fira Sans, sans-serif',
-  iconSize: 14,
-  columnHeaderHeight: 24,
-  filterBarHeight: 24,
-} satisfies IrisGridDensity;
-
-export const SPACIOUS_DENSITY_THEME = {
-  ...NORMAL_DENSITY_THEME,
-  cellHorizontalPadding: 7,
-  headerHorizontalPadding: 15,
-  rowHeight: 28,
-} satisfies IrisGridDensity;
+/**
+ * Get the density specific theme settings for a density.
+ * @param density Density of the theme to get
+ * @returns Density specific theme settings
+ */
+export function getDensityTheme(
+  density: 'compact' | 'regular' | 'spacious'
+): IrisGridDensity {
+  // This must be read when the function is called and not in the global scope of the module
+  // Otherwise it initializes w/ a bunch of empty values
+  const IrisGridTheme = resolveCssVariablesInRecord(IrisGridThemeRaw);
```

Comment:
I think we can simplify this. Instead of having `IrisGridDensity` be explicitly defined, it should just be something like:
```
export interface IrisGridDensity = Partial<IrisGridThemeType>;
```
I don't think we need to get fancier than that, but if we do, add in a `Pick`:
```
export interface IrisGridDensity = Partial<Pick<IrisGridThemeType, 'cellHorizontalPadding' | ...>>;
```

Afterward, this `getDensityTheme` can just return a partial for the properties it does override from the normal theme. So for `'regular'`, you'd just return `{}` since you just want the regular theme, `'compact'` you'd return what you are now, and for `'spacious'` you'd just return:
```
{
    cellHorizontalPadding: 7,
    headerHorizontalPadding: 15,
    rowHeight: 28,
}
```

Then this function doesn't need to resolve the CSS variables and the override is simpler.

Reply:
With the way the global density setting is implemented, I think this wouldn't work for overriding the individual densities.

Right now, the global setting is merged at the `IrisGridThemeContext` to provide the color + density choices as the default for grids. If the densities aren't defined fully, then there's a problem when you change the global setting and also set a density explicitly w/ dh.ui.

For example, set the global density to compact. Then with dh.ui, create a ui.table w/ `density="regular"`. Now we need either a separate context or context value from `IrisGridThemeContext` to apply the global density in `IrisGrid` or we need `IrisGrid` to be able to fetch the actual values for the regular density. But you can't just get the default theme and spread it for default density. Because now you've reverted the color part of the theme set in the context.

So we need to know what to override when `<IrisGrid density="..." />` is used. This seemed like the easiest way. Otherwise `IrisGrid` is dealing with global, app-specific redux state here to fetch the global density instead of just reading its theme context for global theme.

Reply:
Okay, so sleeping on this the thing that's bugging me is that we're arbitrarily defining the density theme overrides; really you should be able to define all the different densities within the theme itself, and then you pick which densities. Otherwise say you wanted to theme the `regular` theme with a default `cellHorizontalPadding: 15`, but you can't theme the `spacious` density so when the user switches to that, it would actually reduce density (making the "Density" options useless, depending on your originally defined theme). So somehow we should be able to use the theme to define all the different densities, and that can be the theme passed in to the ThemeProvider.
Just taking a look at how Spectrum handles this. In their tabs, they are just setting an extra CSS class if the `compact` density is used: https://github.com/adobe/react-spectrum/blob/9b48ec5d528adeba1a96592945459726def7ba56/packages/%40react-spectrum/tabs/src/Tabs.tsx#L297
Same with ListView: https://github.com/adobe/react-spectrum/blob/9b48ec5d528adeba1a96592945459726def7ba56/packages/%40react-spectrum/list/src/ListView.tsx#L232
Then they define the extra CSS for those classes specifically: https://github.com/adobe/react-spectrum/blob/9b48ec5d528adeba1a96592945459726def7ba56/packages/%40react-spectrum/list/src/styles.css#L548

It would make sense to do something similar... though I don't think a nested class definition in `IrisGridTheme.module.scss` actually generates the correct object when imported, it looks like it gets mangled (I tried adding a `.spacious { }` section within the module.scss and imported):
![image](https://github.com/user-attachments/assets/dbcc159a-5846-41cf-b0c1-9169d1f4bcad)

In any case, we can just define it using a `--compact` or `--spacious` suffix as Spectrum does for their class names, and filter out properties based on that suffix for density.
Then the theme defined in the context is defined entirely in SCSS (as it should be) and defines all the different density specs, and then we just pick the appropriate one based on the density provided for a particular grid.
With that in mind, the default density setting should not override the theme itself... really it should be a separate context. However, that's a bit of a pain for `IrisGrid` which is a class component and only has one `contextType`... so then `IrisGridPanel`, `GridWidgetPlugin`, etc. need to read the density from settings and pass in that prop.

Reply:
Moved compact and spacious into the `IrisGridTheme` interface. Didn't bother with CSS variables because almost none of the size is currently in CSS variables, so might as well do that all at once if we want to support grid size properties in the main theme